test/integration/controlplane/synthetic_controlplane_test.go

```diff
@@ -311,29 +349,36 @@ func constructBody(val string, size int, field string, t *testing.T) *appsv1.Dep
 }
 
 func TestObjectSizeResponses(t *testing.T) {
-	_, s, closeFn := framework.RunAnAPIServer(nil)
-	defer closeFn()
-
-	client := clientset.NewForConfigOrDie(&restclient.Config{Host: s.URL})
-
-	const DeploymentMegabyteSize = 100000
-	const DeploymentTwoMegabyteSize = 175000
-	const DeploymentThreeMegabyteSize = 250000
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--storage-media-type=application/json"}, framework.SharedEtcd())
+	defer server.TearDownFn()
+
+	server.ClientConfig.ContentType = runtime.ContentTypeJSON
+	client := clientset.NewForConfigOrDie(server.ClientConfig)
+
+	// Computing ManagedFields is extremely inefficient for large object, e.g.
+	// it may take 10s+ to just compute it if we have ~100k very small labels or
+	// annotations.  This in turn may lead to timing out requests,
+	// which have hardcoded timeout of 34 seconds.
+	// As a result, we're using slightly larger individual labels/annotations
+	// to reduce the number of those.
```

Comment:
should we track this in a different issue?

Reply:
it is a regression on performance, isn't it?

Reply:
We probably should (although I also don't perceive it as very high priority - usecase of tens thousands of labels/annotations isn't something that I'm hearing about as a real-life usecase).

Re regression - it's related to enabling SSA - so it's not a new regression.
[The reason it wasn't a problem before this PR is that by not enabling OpenAPI (which the old framework didn't) it was effectively also disabling SSA...]

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/111215